src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -29,14 +32,38 @@
             Thread.currentThread().interrupt();
             throw new FileSystemExecutionException(op + " - execution interrupted", ex);
         } catch (ExecutionException ex) {
-            // If this exception has a cause that is a runtime exception, then we rethrow the cause.
-            // Typically, the cause is one of our file system messages we want to see.
-            // If there is no cause or it is not runtime, we wrap the exception in a generic one of ours.
+            // The ExecutionException wraps the underlying exception caught in the Future, so we need
+            // to examine the properties of the cause to understand what to do.
+            // Possible outcomes:
+            // - throw FileSystemRetryException for retryable firestore exceptions to ask the step to retry
+            // - throw FileSystemExecutionException for other firestore exceptions (not retryable)
+            // - rethrow RuntimeExceptions to expose other unexpected exceptions
+            // - throw FileSystemExecutionException to wrap non-Runtime (oddball) exceptions
+
             Throwable throwable = ex.getCause();
+            while (throwable instanceof ExecutionException) {
+                ExecutionException eex = (ExecutionException)throwable;
```

Comment:
Is this cast necessary? This code is calling `Throwable.getCause()` so I wouldn't expect to see a cast here.

Reply:
Fixed.